apiextensions/v1alpha1/types_compatibilityrequirement.go

```diff
@@ -0,0 +1,368 @@
+type APIVersionString string
+
+// APIVersions specifies a set of API versions of a CRD.
+// +kubebuilder:validation:XValidation:rule="self.defaultSelection != 'AllServed' || !has(self.additionalVersions)",message="additionalVersions may not be defined when defaultSelection is 'AllServed'"
+type APIVersions struct {
+	// defaultSelection specifies a method for automatically selecting a set of
+	// versions to require.
+	//
+	// Valid options are StorageOnly and AllServed.
+	// When set to StorageOnly, only the storage version is selected for
+	// compatibility assessment.
+	// When set to AllServed, all served versions are selected for compatibility
+	// assessment.
+	//
+	// This field is required.
+	// +required
+	DefaultSelection APIVersionSelectionType `json:"defaultSelection,omitempty"`
+
+	// additionalVersions specifies a set api versions to require in addition to
+	// the default selection. It is explicitly permitted to specify a version in
+	// additionalVersions which was also selected by the default selection. The
+	// selections will be merged and deduplicated.
+	//
+	// Each item must be at most 63 characters in length, and must must consist
+	// of only lowercase alphanumeric characters and hyphens, and must start
+	// with an alphabetic character and end with an alphanumeric character.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=32
+	// +listType=set
+	// +optional
+	AdditionalVersions []APIVersionString `json:"additionalVersions,omitempty"`
+}
+
+// APIExcludedField describes a field in the schema which will not be validated by
+// crdSchemaValidation or objectSchemaValidation.
+type APIExcludedField struct {
+	// path is the path to the field in the schema.
+	// Paths are dot-separated field names (e.g., "fieldA.fieldB.fieldC") representing nested object fields.
+	// Each field name must be a valid Kubernetes CRD field name: start with a letter, contain only
+	// letters, digits, and underscores, and be between 1 and 63 characters in length.
+	// A path may contain at most 16 fields.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=1023
+	// +kubebuilder:validation:XValidation:rule="self.split('.', 16).all(f, f.matches('^[a-zA-Z][a-zA-Z0-9_]{0,62}$'))",message="path must be dot-separated field names, each starting with a letter and containing only letters, digits, and underscores not exceeding 63 characters. There may be at most 16 fields in the path."
+	// +required
+	Path string `json:"path,omitempty"`
+
+	// version is the API version the field is excluded from.
+	// When not specified, the field is excluded from all versions.
+	//
+	// When present, must be at most 63 characters in length, and must must
+	// consist of only lowercase alphanumeric characters and hyphens, and must
+	// start with an alphabetic character and end with an alphanumeric
+	// character.
+	// +optional
+	Version APIVersionString `json:"version,omitempty"`
```

Comment:
Why not make this a list? So I can exclude a particular path from multiple versions at once? (Thinking about a case where I may have 3 or 4 versions served at once - rare I guess)